src/vmod_var.c

```diff
@@ -109,6 +109,11 @@ vh_get_var_alloc(struct var_head *vh, const char *name, const struct vrt_ctx *ct
 	if (!v) {
 		/* Allocate and add */
 		v = (struct var*)WS_Alloc(ctx->ws, sizeof(struct var));
+		if (v == NULL)
+		{
+			VRT_fail(ctx, "var: alloc var: out of workspace");
```

Comment:
bikeshedding, feel free to ignore: maybe just "vmod_var: out of workspace"? My issue is that "var" is a bit generic, and might be confused with other stuff than the vmod, and "alloc var" isn't useful neither for the user nor for the developer, so we can get rid of it.

Reply:
So, this is actually not bikeshedding but a valid point.

We vendored the var vmod in one of our Rust vmods and in that version of the code, we distinguish between failing to allocate the struct var and copying the name of the variable, hence the detailed error message.

I agree that renaming the first "var" to "vmod_var" makes it much more clearer!

I tried adding tests for both behaviors in test06.vtc but couldn't get the test to pass (the second log line never shows up and `logexpect -wait` hangs as a result).

Is there something wrong with the test setup that can be fixed, or should I extract that second test case (copying of the name runs out of workspace) into a test07.vtc file?

Reply:
if `logexpect -wait` gets the chance to hang, that means `c1` returned, and so that you got a failure, that you should see in the logs. My guess is that `b` is a bit too small so `var` can copy it, but then Varnish needs more workspace for a couple of headers and fails with that.

Try making `b` bigger, like `00000000000000000000000000000000000000000000000000000000000000000` and see if you can make it fail. Technically, if `b` is 64 bytes long, the variable allocation must fail.

Reply:
That did the trick! I've squashed and force-pushed all changes.